pvanalytics/tests/features/test_daytime.py

```diff
@@ -182,3 +246,146 @@ def test_daytime_variable(clearsky_january):
+                              " for left-aligned data, 'R' for "
+                              "right-aligned data, or 'C' for "
+                              "center-aligned data.")):
+        daytime.get_sunset(daytime_mask_left_aligned,
+                           data_alignment='M')
+
+
+def test_consistent_modeled_midday_series(daytime_mask_right_aligned,
+                                          daytime_mask_left_aligned,
+                                          daytime_mask_center_aligned,
+                                          modeled_midday_series):
+    # Get sunrise and sunset times for each time series (left, right, center),
+    # calculate the midday point, compared to modeled midday,
+    # and compare that it's consistent across all three series
+    # Right-aligned data
+    right_sunset = daytime.get_sunset(daytime_mask_right_aligned,
+                                      data_alignment='R')
+    right_sunrise = daytime.get_sunrise(daytime_mask_right_aligned,
+                                        data_alignment='R')
+    midday_series_right = right_sunrise + ((right_sunset - right_sunrise)/2)
+    midday_series_right.index = midday_series_right.index.date
+    midday_diff_right = (modeled_midday_series -
+                         midday_series_right.drop_duplicates())
+    # Left-aligned data
+    left_sunset = daytime.get_sunset(daytime_mask_left_aligned,
+                                     data_alignment='L')
+    left_sunrise = daytime.get_sunrise(daytime_mask_left_aligned,
+                                       data_alignment='L')
+    midday_series_left = left_sunrise + ((left_sunset - left_sunrise)/2)
+    midday_series_left.index = midday_series_left.index.date
+    midday_diff_left = (modeled_midday_series -
+                        midday_series_left.drop_duplicates())
+    # Center-aligned data
+    center_sunset = daytime.get_sunset(daytime_mask_center_aligned,
+                                       data_alignment='C')
+    center_sunrise = daytime.get_sunrise(daytime_mask_center_aligned,
+                                         data_alignment='C')
+    midday_series_center = center_sunrise + ((center_sunset -
+                                              center_sunrise)/2)
+    midday_series_center.index = midday_series_center.index.date
+    midday_diff_center = (modeled_midday_series -
+                          midday_series_center.drop_duplicates())
+    assert (midday_diff_right.equals(midday_diff_left) &
+            midday_diff_center.equals(midday_diff_right))
+    # Assert that the difference between modeled midday for midday
+    # center-aligned data (and consequently left- and right-aligned,
+    # which are asserted above as identical to center-aligned data) is less
+    # than 10 minutes/600 seconds (this threshold was generally considered
+    # noise in the time shift detection paper).
+    assert all(midday_diff_center.dt.seconds <= 600)
```

Comment:
```suggestion
    assert all(midday_diff_center.dt.total_seconds().abs() <= 600)
```

I think this is needed to handle negative timedeltas correctly

Reply:
Good catch. Updated with this fix.